src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_TIMES_VALUE = 1.5;
+    private Object[] data;
+    private int size;
+
+    public ArrayList() {
+        this.data = new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (data.length == this.size) {
+            resize();
+        }
+        data[this.size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        size++;
```

Comment:
it's better to increment the size variable at the end of the method, after adding the element